src/Umbraco.Cms.Integrations.Automation.Zapier/Models/Dtos/SubscriptionDto.cs

```diff
@@ -1,16 +1,24 @@
-﻿using Newtonsoft.Json;
+﻿using System;
+
+using Newtonsoft.Json;
 
 namespace Umbraco.Cms.Integrations.Automation.Zapier.Models.Dtos
 {
     public class SubscriptionDto
     {
+        [JsonProperty("id")]
+        public int Id { get; set; }
+
+        [JsonProperty("entityId")]
+        public string EntityId { get; set; }
+
         [JsonProperty("hookUrl")]
         public string HookUrl { get; set; }
 
-        [JsonProperty("contentType")]
-        public string ContentType { get; set; }
-
         [JsonProperty("subscribeHook")]
         public bool SubscribeHook { get; set; }
+
+        [JsonProperty("isFormSubscription")]
+        public bool IsFormSubscription => Guid.TryParse(EntityId, out _);
```

Comment:
Thinking here we can make the package more extensible by removing this and adding in a "Type" field.  It should take an integer, and have 1 for content publishing and 2 for form submissions.  We can then more easily add more in future.

There should be a unique index across the EntityId and Type fields.

When you look up an entity, you should look it up by type and entity Id (just in case in future we had two types with the same Id).

And perhaps you can expose this on the dashboard, so you only need a single table for display.

Reply:
Updated